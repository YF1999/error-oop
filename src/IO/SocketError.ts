import { IErrorArguments } from '../Native';
import { InternalErrorTool } from '../InternalTools';

import { IIOErrorOptions, IOError } from './IOError';

export interface ISocketErrorOptions extends IIOErrorOptions {}

/**
 * Applicable when an error occurs on a socket.
 */
export class SocketError extends IOError {
    public constructor();
    /**
     * @param message The error message that explains the reason for this error.
     */
    public constructor(message: string);
    /**
     * @param message The error message that explains the reason for this error.
     * @param innerError The error that is the cause of the current error. Stack trace will be appended.
     */
    public constructor(message: string, innerError: Error);
    /**
     * @param options The constructor options.
     */
    public constructor(options: ISocketErrorOptions);

    public constructor(...args: IErrorArguments<ISocketErrorOptions>) {
        super(InternalErrorTool.parseErrorArguments([], ...args));
    }
}
