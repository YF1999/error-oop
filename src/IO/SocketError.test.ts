import { SocketError } from './SocketError';

standardTestError(SocketError, 'SocketError', 'error');
