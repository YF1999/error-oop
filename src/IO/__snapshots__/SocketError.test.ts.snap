// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test \`SocketError\` own keys 1`] = `
Array [
  "stack",
]
`;

exports[`Test \`SocketError\` own keys 2`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`SocketError\` own keys 3`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`SocketError\` own keys 4`] = `
Array [
  "stack",
  "message",
]
`;
